pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -94,6 +98,16 @@ func NewNetworkPolicyController(antreaClientGetter agent.AntreaClientProvider,
 	// Wait until appliedToGroupWatcher, addressGroupWatcher and networkPolicyWatcher to receive bookmark event.
 	c.fullSyncGroup.Add(3)
 
+	if c.ofClient != nil && features.DefaultFeatureGate.Enabled(features.AntreaPolicy) {
+		// Register packetInHandler
+		c.ofClient.RegisterPacketInHandler(uint8(openflow.PacketInReasonNP), "networkpolicy", c)
+		// Initiate logger for Antrea Policy audit logging
+		err := initLogger()
+		if err != nil {
+			return nil
```

Comment:
Looks like the func returns `*Controller`, for new controller func maybe `nil` return could work as an error.

Reply:
Thanks for reminding, I forgot this function has single return value. Then we have to check nil in the caller anyway, otherwise `go networkPolicyController.Run(stopCh)` would panic. Since `NewNetworkPolicyController` could fail now I would suggest changing the return value to `(*Controller, error)` as it's more Go and we could log the detailed error in the caller.